core/templates/dev/head/domain/utilities/UrlInterpolationService.js

```diff
@@ -17,166 +17,163 @@
-          new String(urlTemplate) + '\'');
-        return null;
+          'Empty path passed in method.');
       }
 
-      // http://stackoverflow.com/questions/4775722
-      if (!(interpolationValues instanceof Object) || (
-          Object.prototype.toString.call(
-            interpolationValues) === '[object Array]')) {
+      var RESOURCE_PATH_STARTS_WITH_FORWARD_SLASH = /^\//;
+      // Ensure that resourcePath starts with a forward slash.
+      if (!resourcePath.match(RESOURCE_PATH_STARTS_WITH_FORWARD_SLASH)) {
         alertsService.fatalWarning(
-          'Expected an object of interpolation values to be passed into ' +
-          'interpolateUrl.');
-        return null;
+          'Path must start with \'\/\': \'' + new String(resourcePath) +
+          '\'.');
       }
+    };
+
+    var getCachePrefixedUrl = function(resourcePath) {
+      validateResourcePath(resourcePath);
+      return GLOBALS.ASSET_DIR_PREFIX + resourcePath;
+    };
+
+    return {
+      /**
+       * Given a formatted URL, interpolates the URL by inserting values the URL
+       * needs using the interpolationValues object. For example, urlTemplate
+       * might be:
+       *
+       *   /createhandler/resolved_answers/<exploration_id>/<escaped_state_name>
+       *
+       * interpolationValues is an object whose keys are variables within the
+       * URL. For the above example, interpolationValues may look something
+       * like:
+       *
+       *   { 'exploration_id': '0', 'escaped_state_name': 'InputBinaryNumber' }
+       *
+       * If a URL requires a value which is not keyed within the
+       * interpolationValues object, this will return null.
+       */
+      interpolateUrl: function(urlTemplate, interpolationValues) {
+        if (!urlTemplate) {
+          alertsService.fatalWarning(
+            'Invalid or empty URL template passed in: \'' +
+            new String(urlTemplate) + '\'');
+          return null;
+        }
 
-      // Valid pattern: <alphanum>
-      var INTERPOLATION_VARIABLE_REGEX = /<(\w+)>/;
+        // http://stackoverflow.com/questions/4775722
+        if (!(interpolationValues instanceof Object) || (
+            Object.prototype.toString.call(
+              interpolationValues) === '[object Array]')) {
+          alertsService.fatalWarning(
+            'Expected an object of interpolation values to be passed into ' +
+            'interpolateUrl.');
+          return null;
+        }
 
-      // Invalid patterns: <<stuff>>, <stuff>>>, <>
-      var EMPTY_VARIABLE_REGEX = /<>/;
-      var INVALID_VARIABLE_REGEX = /(<{2,})(\w*)(>{2,})/;
+        // Valid pattern: <alphanum>
+        var INTERPOLATION_VARIABLE_REGEX = /<(\w+)>/;
 
-      // Parameter values can only contain alphanumerical characters, spaces,
-      // hyphens or underscores.
-      var VALID_URL_PARAMETER_VALUE_REGEX = /^(\w| |_|-)+$/;
+        // Invalid patterns: <<stuff>>, <stuff>>>, <>
```

Comment:
Is there extra indentation here relative to the previous line? And in the following lines too.